### PR TITLE
Add `instances_from_same_machine` variable in preparation

### PR DESCRIPTION
**Pull request by hackallcode**
Now it's possible to make some tasks using `instances_from_same_machine` variable,
that contain hosts from the same machine as the current host.

For example:

```yaml
- name: 'Collect list of files by machines'
  set_fact:
    backup_files_by_machines: '{{ backup_files_by_machines + hostvars[instance].instance_backup_files }}'
  with_items: '{{ instances_from_same_machine[inventory_hostname] }}'
  loop_control:
    loop_var: instance
  # when: inventory_hostname in single_instances_for_each_machine
```

### CODE REVIEW

**Review by hackallcode — COMMENTED** [review]
Self-review, tests passed: https://github.com/tarantool/ansible-cartridge/pull/369

**Comment by dokshina** [comment]
I'm not sure that we need it. Users can specify machines in the limit.

File: library/cartridge_get_cached_facts.py [hunk 1]
```diff
@@ -90,7 +90,7 @@
         'cartridge_eval_retries',
         'cartridge_eval_delay',
     ],
-    'single_instances_for_each_machine': [
+    'facts_for_each_machine': [
```
Comment (dokshina) [inline library/cartridge_get_cached_facts.py]: `facts_for_machines`? `facts_computed_for_machines`?

File: library/cartridge_get_facts_for_each_machine.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    machines_single_instance = {}
+    machines_play_hosts = {}
+    instances_play_hosts = {}
+
+    for instance_name in sorted(play_hosts):
+        instance_vars = module_hostvars[instance_name]
+        machine_id = get_machine_id(instance_vars, instance_name)
+
+        # Calculate play hosts for each machine
+        machines_play_hosts[machine_id] = machines_play_hosts.get(machine_id, [])
```
Comment (dokshina) [inline library/cartridge_get_facts_for_each_machine.py]: `play_hosts_by_machines`?

File: defaults/main.yml [hunk 3]
```diff
@@ -178,8 +178,9 @@ alive_not_expelled_instance: null
 
 cached_facts_res: null
 cached_facts: null
-single_instances_for_each_machine_res: null
+facts_for_each_machine_res: null
 single_instances_for_each_machine: null
+play_hosts_from_same_machine: null
```
Comment (dokshina) [inline defaults/main.yml]: Maybe `instances_from_same_machine`?

File: doc/steps.md [hunk 4]
```diff
@@ -66,6 +66,7 @@ Some of useful variables always establishes during role preparation, so them can
   - `dirs_to_remove_on_cleanup` - dirs that will be removed on instance cleanup;
 - `single_instances_for_each_machine` - list of instances (Ansible hosts), one for each physical machine,
   for example, can be used in `delegate_to`;
+- `instances_from_same_machine` - list of instances that on the same physical machine;
```
Comment (dokshina) [inline doc/steps.md]: Seems that it's a map, not a list.

Comment (dokshina) [reply doc/steps.md]: Maybe add a little example? Like "Use `instances_from_same_machine[inventory_hostname]` to get a list of instances placed on the same machine as the current one".

**Review by dokshina — DISMISSED** [review]

**Review by dokshina — APPROVED** [review]

**Pull request merged**